### PR TITLE
e2e fixes - part 0

### PR DESCRIPTION
**Pull request by runcom**
First batch of fixes needed for k8s e2e tests. Most importantly this came out https://github.com/kubernetes-incubator/cri-o/issues/531

See individual commits messages for rationales behind these patches.

@mrunalp @sameo PTAL

### CODE REVIEW

**Comment by runcom** [comment]
seems like the latest commit is failing on Travis because after a CRI-O restart, image IDs changes :confused: that shouldn't be the case, I'll comment something and report the bug in #531

**Comment by runcom** [comment]
@mrunalp @sameo this is good to go now, I've re-kicked node-e2e also

File: server/sandbox_stop.go [hunk 1]
```diff
@@ -16,7 +16,13 @@ func (s *Server) StopPodSandbox(ctx context.Context, req *pb.StopPodSandboxReque
 	logrus.Debugf("StopPodSandboxRequest %+v", req)
 	sb, err := s.getPodSandboxFromRequest(req.PodSandboxId)
 	if err != nil {
-		return nil, err
+		if err == errSandboxIDEmpty {
+			return nil, err
+		}
+
+		resp := &pb.StopPodSandboxResponse{}
+		logrus.Warnf("could not get sandbox %s, it's probably been removed already: %v", req.PodSandboxId, err)
```
Comment (runcom) [inline server/sandbox_stop.go]: s/removed/stopped

**Comment by runcom** [comment]
retest

**Comment by runcom** [comment]
ok to go

**Comment by mrunalp** [comment]
Should we wait for #531 to be fixed for the image related changes?

**Comment by runcom** [comment]
Nope, #531 isn't really blocking and not breaking any e2e test. We should fix #531 by v0.5 I'd say.

File: server/sandbox_stop.go [hunk 2]
```diff
@@ -16,7 +16,13 @@ func (s *Server) StopPodSandbox(ctx context.Context, req *pb.StopPodSandboxReque
 	logrus.Debugf("StopPodSandboxRequest %+v", req)
 	sb, err := s.getPodSandboxFromRequest(req.PodSandboxId)
 	if err != nil {
-		return nil, err
+		if err == errSandboxIDEmpty {
```
Comment (mrunalp) [inline server/sandbox_stop.go]: Can we add a comment that we are always returning nil to kubelet for Stop?

**Comment by mrunalp** [comment]
nit: 3rd commit seems to be modifying the 2nd commit. Can we either clean those up or squash them?

File: server/sandbox_stop.go (hunk 2 above)
Comment (runcom) [inline server/sandbox_stop.go]: Mmm where? We don't have any comment even in Remove

Comment (runcom) [inline server/sandbox_stop.go]: we aren't _always_ returning nil to kubelet for stop, we return err if the sandboxId is empty. I guess I haven't understood your comment :) could you clarify?

**Comment by runcom** [comment]
> nit: 3rd commit seems to be modifying the 2nd commit. Can we either clean those up or squash them?

cleaned 2nd and 3rd commits

File: server/sandbox_stop.go (hunk 2 above)
Comment (mrunalp) [inline server/sandbox_stop.go]: I misread the code :)

File: server/sandbox_stop.go [hunk 3]
```diff
@@ -16,7 +16,13 @@ func (s *Server) StopPodSandbox(ctx context.Context, req *pb.StopPodSandboxReque
...
+			return nil, err
+		}
+
+		resp := &pb.StopPodSandboxResponse{}
+		logrus.Warnf("could not get sandbox %s, it's probably been stopped already: %v", req.PodSandboxId, err)
+		return resp, nil
```
Comment (mrunalp) [inline server/sandbox_stop.go]: I think this section deserves a comment.

Comment (runcom) [inline server/sandbox_stop.go]: should we add a comment in `server/sandbox_remove.go` as well then?

Comment (mrunalp) [inline server/sandbox_stop.go]: Yeah, let us add comments so it is clear that we are trying to adhere to kubelet's current expectations.

**Comment by runcom** [comment]
@mrunalp updated, added comments.

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**